app/controllers/concerns/auth_source_sso.rb

```diff
@@ -110,7 +110,7 @@ def save_user!(user)
   end
 
   def sso_in_progress!
-    sso_failure_in_progress! || session[:auth_source_registration]
+    sso_failure_in_progress! || session[:auth_source_registration] || session[:authenticated_user_id]
```

Comment:
This is a problem as this remembers the user in the session. So changes to the `auth_source_sso` header will be ignored for the duration of the session.

This was probably intended as to not trigger the stages with every single request but we need to find another way to achieve this. Also we should add specs for this scenario. Testing that the auth stages will only be triggered once per session IFF the header doesn't change.

We could store the header value in the session too and make sure to reset everything if it has changed to mitigate this issue.

Reply:
This line doesn't change anything about the user in the session, that was always handled in the `super` call of the `setup_user` here: https://github.com/opf/openproject/blob/dev/app/controllers/concerns/auth_source_sso.rb#L7 If there is an active user, thatis always returned first

Reply:
Right, my bad! all good then with this PR! The issue I mentioned then has nothing to do with your PR at all but is still something we should think about fixing, I think. Although it's an unlikely scenario so it's a low priority.

Either way this PR is good to merge then. Thanks. 👍